File: Leonardo/teste.py
def calcula_graos():
    total_graos = 0
    graos_no_quadro = 1

    for linha in range(1, 9):
        for coluna in range(1, 9):
            total_graos += graos_no_quadro
            graos_no_quadro *= 2

    return total_graos

# Chamando a função e exibindo o resultado
resultado = calcula_graos()
print("O número total de grãos que o monge esperava receber é:", resultado)

File: Leonardo/teste.c
#include <stdio.h>

int main() {
    int linhas = 8;
    int colunas = 8;
    int quadros = linhas * colunas;
    long long int graos = 1;  // Utilizamos um tipo long long int para lidar com números grandes
    long long int total_graos = 0;
    
    for (int i = 1; i <= quadros; i++) {
        total_graos += graos;
        graos *= 2;
    }
    
    printf("O monge esperava receber um total de %lld grãos.\n", total_graos);
    
    return 0;
}